gui/src/main/loading-bar.tsx

```diff
@@ -0,0 +1,18 @@
+import { AuthMachineContext } from "@/machines/auth-machine"
+import { useSelector } from "@xstate/react"
+
+export const LoadingBar = () => {
+  const authActor = AuthMachineContext.useActorRef();
+  const loaded = useSelector(authActor, ({ context }) => context.loaded);
```

Comment:
How come you decided on this approach as opposed to injecting the state?

Reply:
Just to keep it consistent, since we don't inject the state anywhere else